### DISCUSSION TITLE
Controller+gateway in the same instance?

### DISCUSSION THREAD

**Discussion by squromiv** [dec0dOS/zero-ui#95] [Q&A]
Launched zero-ui in rented VPS with static IP. 
I have read some info about [full tunnel mode](https://zerotier.atlassian.net/wiki/spaces/SD/pages/7110693/Overriding+Default+Route+Full+Tunnel+Mode). So is it possible to get a gateway for this mode in the VPS where zero-ui is already launched? It is not the best way - to pay extra money for another server .

**Comment by dec0dOS** [comment]
@squromiv. It should work as expected. In docker-compose configuration it may be necessary to use the [network host mode](https://docs.docker.com/compose/compose-file/compose-file-v3/#network_mode).

**Comment by squromiv** [comment]
I'm totally new to Docker so I appreciate your patience. Should I use the same [container](https://hub.docker.com/r/zyclonite/zerotier/) as used in zero-ui? Will it work as "client"? What should I change in [docker-compose.yml](https://github.com/zyclonite/zerotier-docker/blob/main/docker-compose.yml) in order to run them simultaneously?

**Reply by dec0dOS** [reply]
You could use your controller node as an ordinary client (you have to join the network first) and change the docker-compose network mode to host. 
Or you could run the second zerotier-one instance outside the docker.

**Reply by squromiv** [reply]
> you have to join the network first

Launched a command "zerotier-cli join" from container. controller node is placed now in network, but offline. Looks like [this](https://discuss.zerotier.com/t/zerotier-cli-join-500-error/6610) error.
![1](https://user-images.githubusercontent.com/15985877/176953767-339380f2-8bf8-4cd6-bb7b-010aef9e0fc9.jpg)
![3](https://user-images.githubusercontent.com/15985877/176954239-3edb8016-e8b3-4cb5-9771-1c9d3c587999.jpg)
![4](https://user-images.githubusercontent.com/15985877/176954906-21738313-2aff-450f-bac8-f4544e6281d4.jpg)
![2](https://user-images.githubusercontent.com/15985877/176953771-69d7bf6b-597c-4d87-b7d7-2a6bab94b3ce.jpg)

**Reply by dec0dOS** [reply]
There is a bug in ZeroUI with the controller status. https://github.com/dec0dOS/zero-ui/issues/31
Other nodes should be displayed correctly

**Reply by squromiv** [reply]
> There is a bug in ZeroUI with the controller status

Hm. I deleted zero-ui member from network. Tried to join it again. And it is not visible now at all in ZeroUI interface. Restarted docker compose. Same result. Is this another bug?
